src/prototype_creep_fight.js

```diff
@@ -342,128 +342,357 @@ Creep.prototype.fightRanged = function(target) {
+      this.moveTo(constructionSites, {
+        reusePath: 0,
+        ignoreCreeps: true
+      });
       return true;
     }
+    this.moveTo(structures, {
+      reusePath: 0,
+      ignoreCreeps: true
+    });
+    this.dismantle(structures);
+    return true;
+  }
+
+  let search = PathFinder.search(
+    this.pos, {
+      pos: spawn.pos,
+      range: 1
+    }, {
+      maxRooms: 1
+    }
+  );
+
+  var healer = this.pos.findInRange(FIND_MY_CREEPS, 45, {
+    filter: function(object) {
+      return object.memory.role == 'squadheal';
+    }
+  });
+
+  var attackerClose = this.pos.findInRange(FIND_MY_CREEPS, 5, {
+    filter: function(object) {
+      return object.memory.role == 'squadsiege' || object.memory.role == 'autoattackmelee';
+    }
   });
-  let target = tower;
-  if (tower === null) {
-    var spawn = this.pos.findClosestByRange(FIND_HOSTILE_STRUCTURES, {
+
+  var attackerFar = this.pos.findInRange(FIND_MY_CREEPS, 48, {
+    filter: function(object) {
+      return object.memory.role == 'squadsiege' || object.memory.role == 'autoattackmelee';
+    }
+  });
+
+  if (this.pos.getRangeTo(spawn.pos) <= 1) {
+    this.dismantle(spawn);
+    return true;
+  } else if (search.path[0].lookFor(LOOK_STRUCTURES).length !== 0 && search.path[0].lookFor(LOOK_STRUCTURES)[0].structureType != STRUCTURE_ROAD) {
+    this.say('RazingLow!', true);
+    let structures = this.pos.findInRange(FIND_STRUCTURES, 1, {
       filter: function(object) {
-        if (object.structureType == 'spawn') {
-          return true;
-        }
-        return false;
+        return object.structureType != STRUCTURE_ROAD;
       }
     });
-    target = spawn;
+    structures.sort(function(a, b) { return a.hits - b.hits; });
+    this.dismantle(structures[0]);
+    this.moveTo(structures[0], {
+      reusePath: 0,
+      ignoreCreeps: true
+    });
+    return true;
+  } else if (search.path[0].lookFor(LOOK_CREEPS).length !== 0 && search.path[1] && search.path[1].lookFor(LOOK_STRUCTURES).length !== 0 && search.path[1].lookFor(LOOK_STRUCTURES)[0].structureType != STRUCTURE_ROAD) {
+    this.say('Razing!', true);
+    this.moveTo(search.path[1], {
+      reusePath: 0
+    });
+    return true;
+  } else if (healer.length > 2 || (attackerClose === 0 && attackerFar !== 0) || !this.memory.squad) {
+    this.say('Terminatin', true);
+    let structures = this.pos.findInRange(FIND_STRUCTURES, 1, {
+      filter: function(object) {
+        return object.structureType != STRUCTURE_ROAD;
+      }
+    });
+    structures.sort(function(a, b) { return a.hits - b.hits; });
+    this.dismantle(structures[0]);
+    this.move(this.pos.getDirectionTo(search.path[0]), {
+      ignoreCreeps: true
+    });
+    return true;
   }
-  if (target === null) {
-    var cs = this.pos.findClosestByRange(FIND_CONSTRUCTION_SITES);
-    this.moveTo(cs);
-    return false;
+  if (this.avoidEdge(2)) {
+    return true;
+  }
+  return false;
+};
+
+Creep.prototype.autosiege = function() {
```

Comment:
shouldn't the two functions be named `autodismantle` and `autoattack` or similar.

Maybe even combine both and check for bodyparts instead.